agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/reminder/ReminderDetailState.kt

```diff
@@ -0,0 +1,30 @@
+package com.pronoidsoftware.agenda.presentation.detail.reminder
+
+import com.pronoidsoftware.agenda.presentation.detail.model.NotificationDuration
+import com.pronoidsoftware.core.domain.util.now
+import com.pronoidsoftware.core.domain.util.today
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
+
+data class ReminderDetailState(
+    val clock: Clock = Clock.System,
```

Comment:
Does this change? If yes, in which case? And how does it have an impact on the UI? Doesn't feel like a state to me 🤔

Reply:
I'm only supplying this here in order to replace it for testing. Right now, I'm injecting the clock into the VM and can easily replace the module that provides it that way, and the VM updates the state with it. But I agree that it's not really UI state, so perhaps I should hold onto it as a variable within the VM? I'm unsure if there is a nice way to use hilt to inject into composables since they're functions, but that would be nice here...

Reply:
Refactored here: https://github.com/evanadwyer-pronoid-software/Tasky/pull/69